gcovr/cobertura_xml_generator.py

```diff
@@ -174,47 +197,31 @@ def print_xml_report(covdata, options):
     for packageName in keys:
         packageData = packages[packageName]
         package = packageData[0]
-        packageXml.appendChild(package)
-        classes = doc.createElement("classes")
-        package.appendChild(classes)
+        packageXml.append(package)
+        classes = etree.SubElement(package, "classes")
         classNames = list(packageData[1].keys())
         classNames.sort()
         for className in classNames:
-            classes.appendChild(packageData[1][className])
-        package.setAttribute("name", packageName.replace(os.sep, '.'))
-        package.setAttribute(
+            classes.append(packageData[1][className])
+        package.set("name", packageName.replace(os.sep, '.'))
+        package.set(
             "line-rate", str(packageData[2] / (1.0 * packageData[3] or 1.0))
         )
-        package.setAttribute(
+        package.set(
             "branch-rate", str(packageData[4] / (1.0 * packageData[5] or 1.0))
         )
-        package.setAttribute("complexity", "0.0")
+        package.set("complexity", "0.0")
 
     # Populate the <sources> element: this is the root directory
-    source = doc.createElement("source")
-    source.appendChild(doc.createTextNode(options.root.strip()))
-    sources.appendChild(source)
-
-    if options.prettyxml:
-        import textwrap
-        lines = doc.toprettyxml(" ").split('\n')
-        for i in xrange(len(lines)):
-            n = 0
-            while n < len(lines[i]) and lines[i][n] == " ":
-                n += 1
-            lines[i] = "\n".join(textwrap.wrap(
-                lines[i], 78,
-                break_long_words=False,
-                break_on_hyphens=False,
-                subsequent_indent=" " + n * " "
-            ))
-        xmlString = "\n".join(lines)
-        # print textwrap.wrap(doc.toprettyxml(" "), 80)
-    else:
-        xmlString = doc.toprettyxml(indent="")
-    if options.output is None:
-        sys.stdout.write(xmlString + '\n')
-    else:
-        OUTPUT = open(options.output, 'w')
-        OUTPUT.write(xmlString + '\n')
-        OUTPUT.close()
+    etree.SubElement(sources, "source").text = options.root.strip()
+
+    with smart_open(options.output) as fh:
+        if LXML_AVAILABLE:
+            print >>fh, \
+                etree.tostring(root,
+                               pretty_print=options.prettyxml,
+                               encoding="UTF-8",
+                               xml_declaration=True,
+                               doctype="<!DOCTYPE coverage SYSTEM 'http://cobertura.sourceforge.net/xml/coverage-04.dtd'>")
+        else:
+            print >>fh, etree.tostring(root, encoding="UTF-8")
```

Comment:
should use `fh.write(...)` or the print function `print(..., file=fh)`. Print statements only work in Python 2.